src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// The exception list to string converter.
+    /// </summary>
+    public static class ExceptionListToStringConverter
+    {
+        /// <summary>
+        /// Extracts correlation IDs from the list of exceptions.
+        /// </summary>
+        /// <param name="exceptions">List of exceptions from which correlation IDs are extracted.</param>
+        /// <returns>List of correlation IDs.</returns>
+        public static List<string> ExtractCorrelationIDsFromException(IList<Exception> exceptions)
+        {
+            var correlationIDs = new List<string>();
+
+            if (exceptions == null)
+            {
+                return null;
+            }
+
+            foreach (Exception exception in exceptions)
+            {
+                if (exception.GetType() == typeof(MsalServiceException))
+                {
+                    var msalServiceException = (MsalServiceException)exception;
+                    if (msalServiceException.CorrelationId != null)
+                    {
+                        correlationIDs.Add(msalServiceException.CorrelationId);
+                    }
+                }
+                else if (exception.GetType() == typeof(MsalUiRequiredException))
+                {
+                    var msalUiRequiredException = (MsalUiRequiredException)exception;
+                    if (msalUiRequiredException.CorrelationId != null)
+                    {
+                        correlationIDs.Add(msalUiRequiredException.CorrelationId);
+                    }
+                }
+            }
+
+            return correlationIDs;
+        }
+
+        /// <summary>
+        /// Executes the convertion of exceptions to a JSON string.
+        /// </summary>
+        /// <param name="exceptions">List of exceptions.</param>
+        /// <returns>JSON format of exception list.</returns>
+        public static string Execute(IEnumerable<Exception> exceptions)
+        {
+            if (exceptions == null || exceptions.Count() == 0)
+            {
+                return null;
+            }
+
+            List<SerializableException> customExceptions = new List<SerializableException>();
+            foreach (Exception exception in exceptions)
+            {
+                var customException = new SerializableException(exception);
+                customExceptions.Add(customException);
+            }
+
+            return JsonSerializer.Serialize(customExceptions);
+        }
+    }
+
+    /// <summary>
+    /// CustomException class with only select properties of Exception class.
+    /// </summary>
+    internal class SerializableException

```

Comment:
I think this can be a `private` class _on_ the `ExceptionListToStringConverter` itself rather than just in the same namespace.

Reply:
After further consideration, does this even need to be a class? Could it just be a private method instead? 🤔

Reply:
Good point, but I don't think so, Tucker. Because the JSON serialization libraries work on a class. If we write a method, we will have to go by our own serialization method as well.